src/devices/sound/gb.cpp

```diff
@@ -411,10 +401,10 @@ void gameboy_sound_device::update_square_channel(SOUND &snd, uint64_t cycles)
 		{
 			cycles -= distance;
 			distance = 0x800 - snd.frequency;
-			uint64_t counter = 1 + cycles / distance;
+			const uint64_t counter = 1 + cycles / distance;
 
 			snd.duty_count = (snd.duty_count + counter) & 0x07;
-			snd.signal = wave_duty_table[snd.duty][snd.duty_count];
+			snd.signal = BIT(wave_duty_table[snd.duty], snd.duty_count);
```

Comment:
This changes the output for these channels from balanced -1/+1 to unbalanced 0/+1 levels.  Is that correct?  Can you provide a reference for this change in behaviour?

Reply:
That is digital sample, and converted at DAC output.